Integrations/src/main/java/io/deephaven/integrations/learn/Computer.java

```diff
@@ -0,0 +1,81 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.db.tables.Table;
+import io.deephaven.db.v2.sources.ColumnSource;
+import org.jpy.PyObject;
+
+/**
+ * Passes indices to Future for deferred calculation.
+ */
+public class Computer {
+
+    private final PyObject modelFunc;
+    private final int batchSize;
+    private final Input[] inputs;
+    private final ColumnSource<?>[][] colSet;
+    private Future current;
+    private int offset;
+
+    /**
+     * Creates a new Computer.
+     *
+     * @param modelFunc     function to use for AI training / prediction on the given inputs.
+     * @param inputs        inputs to the model function.
+     * @param batchSize     maximum number of rows for deferred computation.
+     */
+    public Computer(Table table, PyObject modelFunc, int batchSize, Input ... inputs) {
+
+        if (modelFunc == null) {
+            throw new IllegalArgumentException("Model function cannot be null.");
+        }
+
+        if (batchSize <= 0) {
+            throw new IllegalArgumentException("Max size must be a strictly positive integer.");
+        }
+
+        if (inputs.length == 0) {
+            throw new IllegalArgumentException("Cannot have an empty input list.");
+        }
+
```

Comment:
1) What if the contents of the input array are null?  This case isn't being checked.
2) Given that this is deep in the code and not user exposed, I don't know that varargs are useful.  I think they are actually making the parameter ordering worse.  Consider `Computer(Table table, PyObject modelFunc, Input[] inputs, int batchSize)`.